### PR TITLE
Feature/3190 remove esprit

### PR DESCRIPTION
**Pull request by amdomanska**
# Remove esprit

See https://github.com/DOAJ/doajPM/issues/3190

Remove esprit from DOAJ project (apart from in legacy scripts which will no longer run) in favour of the Elasticsearch python library and our `es_connection` singleton.

## Categorisation

This PR...
- [ ] has scripts to run
- [ ] has migrations to run
- [ ] adds new infrastructure
- [x] changes the CI pipeline
- [ ] affects the public site
- [ ] affects the editorial area
- [ ] affects the publisher area
- [ ] affects the monitoring

## Basic PR Checklist

Instructions for developers:
* For each checklist item, if it is N/A to your PR check the N/A box
* For each item that you have done and confirmed for yourself, check Developer box (including if you have checked the N/A box)

Instructions for reviewers:
* For each checklist item that has been confirmed by the Developer, check the Reviewer box if you agree
* For multiple reviewers, feel free to add your own checkbox with your github username next to it if that helps with review tracking

### Code Style

- No deprecated methods are used
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer

- No magic strings/numbers - all strings are in `constants` or `messages` files
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- ES queries are wrapped in a Query object rather than inlined in the code
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- Where possible our common library functions have been used (e.g. dates manipulated via `dates`)
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- Cleaned up commented out code, etc
  - [ ] N/A
  - [ ] Developer
  - [x] Reviewer

### Testing

- Unit tests have been added/modified
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer
  
- Functional tests have been added/modified
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer
  
- Code has been run manually in development, and functional tests followed locally
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer

### Documentation

- FeatureMap annotations have been added
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- Documentation updates - if needed - have been identified and prepared for inclusion into main documentation (e.g. added and highlighted/commented as appropriate to this PR)
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer
  
- Core model documentation has been added to if needed: https://docs.google.com/spreadsheets/d/1lun2S9vwGbyfy3WjIjgXBm05D-3wWDZ4bp8xiIYfImM/edit
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer

- Events and consumers documentation has been added if needed: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer
  
- The docs for this branch have been generated and pushed to the doc site (see docs/README.md for details)
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer

### Release Readiness

- If needed, migration has been created and tested locally
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer

- Release sheet has been created, and completed as far as is possible https://docs.google.com/spreadsheets/d/1Bqx23J1MwXzjrmAygbqlU3YHxN1Wf7zkkRv14eTVLZQ/edit
  - [ ] N/A
  - [ ] Developer
  - [ ] Reviewer

- There has been a recent merge up from `develop` (or other base branch).  List the dates of the merges up from develop below
  - 08/04/2024

## Testing

List the Functional Tests that must be run to confirm this feature

We need to do the following tests on the test server before release to production:

1. Confirm an Anon Import is working
2. Run a task to check the latest ES backup
3. Run a task to request an ES backup

### Configuration changes

**N/A**

### Scripts
**N/A**

### Migrations

**N/A**

### Monitoring

**N/A**

### New Infrastructure
**N/A**

### Continuous Integration

CI will no longer need to checkout esprit dependency. Handled by changes in `setup.py`.

### CODE REVIEW

**Comment by amdomanska** [comment]
Deleted scripts:

1303_nonexistent_editors_assigned

File: portality/migrate/delete_field_from_type/scrub_field.py [hunk 1]
```diff
@@ -9,23 +9,23 @@
...
 
-def scrub_field(connection, o_type, o_field, batch_size=500):
+def scrub_field(o_type, o_field, batch_size=500):
 
     batch = []
 
-    for o in esprit.tasks.scroll(connection, o_type):
-        # Access the data using the DataObj class
-        o_obj = dataobj.DataObj(raw=o)
+    model_class = MODELS.get(o_type.get("type"))
```
Comment (amdomanska) [inline portality/migrate/delete_field_from_type/scrub_field.py:16]: Is that correct? I wasn't sure how to change o_type to model_class.

**Comment by amdomanska** [comment]
Esprit removed from everywhere where the esprit scroll method could be replaced with dao.scroll. 
All migrations and scripts either updated or deleted if appropriate (for list of deleted files see the comment above).

5 files left that use esprit in different way:

clonelive.py : `esprit.tasks.copy`
discovery.py : `esprit.raw.unpack_json_result`
mappings.py : `esprit.raw.put_mapping` (that is migration)
anon_import.py : `esprit.tasks.bulk_load`

**Comment by RK206** [comment]
@amdomanska Tried to execute 'anon-import' script. It was throwing errors.
```Obtaining account from storage
Retrieved account.bulk.1 from storage
Unzipping account.bulk.1 in temporary store
Importing from account.bulk.1
Traceback (most recent call last):
  File "/Users/rama/CottageLabs/DOAJ/review-code/portality/scripts/anon_import.py", line 110, in <module>
    do_import(config)
  File "/Users/rama/CottageLabs/DOAJ/review-code/portality/scripts/anon_import.py", line 84, in do_import
    imported_count = dao.bulk_load_from_file(uncompressed_file,
  File "/Users/rama/CottageLabs/DOAJ/review-code/portality/dao.py", line 712, in bulk_load_from_file
    ES.bulk(body=f, index=cls.index_name(), doc_type=cls.doc_type())
  File "/Users/rama/CottageLabs/venv/review-code/lib/python3.10/site-packages/elasticsearch/client/utils.py", line 168, in _wrapped
    return func(*args, params=params, headers=headers, **kwargs)
  File "/Users/rama/CottageLabs/venv/review-code/lib/python3.10/site-packages/elasticsearch/client/__init__.py", line 463, in bulk
    return self.transport.perform_request(
  File "/Users/rama/CottageLabs/venv/review-code/lib/python3.10/site-packages/elasticsearch/transport.py", line 415, in perform_request
    raise e
  File "/Users/rama/CottageLabs/venv/review-code/lib/python3.10/site-packages/elasticsearch/transport.py", line 381, in perform_request
    status, headers_response, data = connection.perform_request(
  File "/Users/rama/CottageLabs/venv/review-code/lib/python3.10/site-packages/elasticsearch/connection/http_urllib3.py", line 275, in perform_request
    self._raise_error(response.status, raw_data)
  File "/Users/rama/CottageLabs/venv/review-code/lib/python3.10/site-packages/elasticsearch/connection/base.py", line 330, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(
elasticsearch.exceptions.RequestError: RequestError(400, 'illegal_argument_exception', 'Malformed action/metadata line [3], expected START_OBJECT or END_OBJECT but found [VALUE_STRING]')

Process finished with exit code 1
```
Do I need to do anything to run the script?

**Review by richard-jones — APPROVED** [review]
This looks fine, just requires some tests run on the test server before deploy.

Also the unit tests are not passing in CI, I have not run them locally yet, but if we can get the build to pass that would be sufficient.

**Pull request closed without merging**